diesel/src/query_builder/delete_statement.rs

```diff
@@ -11,12 +11,16 @@ impl<T> DeleteStatement<T> {
 
 impl<T> QueryFragment for DeleteStatement<T> where
     T: UpdateTarget,
+    T::WhereClause: QueryFragment,
 {
     fn to_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
         out.push_context(Context::Delete);
         out.push_sql("DELETE FROM ");
         try!(self.0.from_clause(out));
-        try!(self.0.where_clause(out));
+        if let Some(clause) =  self.0.where_clause() {
```

Comment:
Yeah, it'd probably be better in this case but I don't think it's worth the
git churn to change it.

On Thu, Jan 21, 2016 at 2:37 PM Mike Piccolo <notifications@github.com>
wrote:

> In diesel/src/query_builder/delete_statement.rs
> <https://github.com/sgrif/diesel/pull/122#discussion_r50468280>:
>
> >  {
> >      fn to_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
> >          out.push_context(Context::Delete);
> >          out.push_sql("DELETE FROM ");
> >          try!(self.0.from_clause(out));
> > -        try!(self.0.where_clause(out));
> > +        if let Some(clause) =  self.0.where_clause() {
>
> Fair enough. I am not a big fan of Tuple Structs because it seems like
> normal structs could make the code more readable than the implicit integer
> names.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sgrif/diesel/pull/122/files#r50468280>.
>